Level 2

chyVacheck opened the pull request.

chyVacheck committed:
Create normalize.css

chyVacheck committed:
add icons

chyVacheck committed:
feat: добавить react-router-dom для маршрутизации

chyVacheck committed:
feat: добавить базовое значение фона

chyVacheck committed:
create component logo for header

chyVacheck committed:
create component navigation for header

chyVacheck committed:
create component buttonProfile for header

chyVacheck committed:
create component header

chyVacheck committed:
Create CurrentUserContext.js

chyVacheck committed:
Create Constants.js

chyVacheck committed:
feat add Header, add routers in main App

1. Добавить маршрутизацию (пока что пусто)
2. Подключить компонент Heder

chyVacheck committed:
feat: add local var of color + enable app

chyVacheck committed:
feat: add font 'Inter'

chyVacheck committed:
fix: поменять язык страницы

chyVacheck committed:
feat: add images

chyVacheck committed:
feat: create new component Footer

chyVacheck committed:
feat: create new components TitleOfBlock, LandingBlock

chyVacheck committed:
Update Constants.js

chyVacheck committed:
Update index.css

chyVacheck committed:
feat: feat: create new component Main

1. Создать компонент-стр Main
2. Создать все компоненты для Main

chyVacheck committed:
Update Navigation.jsx

chyVacheck committed:
style: add coments

chyVacheck committed:
refactor: удалить не нужную часть кода

chyVacheck committed:
feat: add component Footer and Main

1. Добавить компоненты Footer и Main в App

chyVacheck committed:
Update index.css

chyVacheck committed:
refactor: перенести функцию из Header в Logo

Раньше Logo принимал функцию которая срабатывала при нажатии на изображение, сейчас же эта функция встроенная, так как оно всегда ведет на главную страницу

chyVacheck committed:
feat: сделать логику для не отрисовки Footer

На страницах /signin /signup Footer, не показывается

chyVacheck committed:
feat: create component SignForm

Компонент SignForm, принимает данный для отрисовки полей ввода, кнопки submit формы и текст

В будущем будет принимать функции для лайв-валидации, а так же для отправки формы на сервер

chyVacheck committed:
feat: create page Register and Login

chyVacheck committed:
refactor: вынести портфолио в отдельный комонент

1. Удалить верстку и стили из компонента AboutMe
2. Создать компонент Portfolio
3. Подключить компонент Portfolio в компоненте AboutMe

chyVacheck committed:
refactor: поменять отображение header и footer

1. Компоненты header и footer отображаются на конкретных стр, раньше на конкретных НЕ отображались
Это нужно для того, что бы на неправильных адресных путях не отрисовывать их
2. Убрана не нужная проверка в компонентах Navigation и ButtonProfile

chyVacheck committed:
feat: добавить страницу 404

1. Добавить стр 404
2. Добавить редирект на нее, при роуте, который не описан

chyVacheck committed:
fix: сделать выравнивание Logo

chyVacheck committed:
feat: сделать адаптив компоненту Promo

chyVacheck committed:
feat: сделать адаптив компоненту Header

chyVacheck committed:
update style: добавить переменные цветов в файл index.css

chyVacheck committed:
add adaptive: component Techs

chyVacheck committed:
add adaptive: component Portfolio

chyVacheck committed:
add adaptive: component AboutMe

chyVacheck committed:
add adaptive: component Footer

chyVacheck committed:
add adaptive: component TitleOfBlock

chyVacheck committed:
add adaptive: component PageNotFound

chyVacheck committed:
add adaptive: component NavTab

chyVacheck committed:
add adaptive: component Navigation

chyVacheck committed:
add adaptive: component LandingBlock

chyVacheck committed:
add adaptive: component AboutProject

chyVacheck committed:
style code: app.js

chyVacheck committed:
style code: component Main

chyVacheck committed:
feat + style code: сделать универсальным Navigation

Теперь компонент подходит для отображения и в шапку профиля и для бугрерного меню, за счет проверки данных из файла констант
+ небольшая стилизация constant.js
+ стили Navigation под бугрерного меню

chyVacheck committed:
feat: сделать универсальным ButtonProfile

Теперь компонент отображается корректно в бургерном меню и в шапке профиля
+ адаптив

chyVacheck committed:
add component: BurgerMenuButton

Кнопка для открытия/закрытия бургерного меню

chyVacheck committed:
add component: BurgerMenu

Компонент бургерное меню, при разрешении 1024 и ниже при авторизированном пользователе появляется возможность открыть данное меню

chyVacheck committed:
style: убрать жесткое позиционирование через padding

chyVacheck committed:
feat + style code: добавить кнопку открытия бургерного меню

Теперь есть кнопка открытия бургерного меню, которая появляется при разрешении 1024 и ниже, при условии что пользователь авторизован

chyVacheck committed:
add new style: добавить цвет текста в бургерном меню

chyVacheck committed:
style: добавить позицию родитель

chyVacheck committed:
feat: добавить BurgerMenu везде + передать пропсы шапке

1. В шапку профиля передать необходимые пропсы, для функционала открытия/закрытия бургерного меню
2. Добавить бургерное меню

chyVacheck committed:
fix: сделать возврат на предыдущую страницу

Сделать возврат на предыдущую страницу, вместо главной

chyVacheck committed:
refactor style: поменять название переменной

chyVacheck committed:
refactor + add: добавить esclint и prettier

Отформатировать код по одним и тем же правилам для поддержания стиля

chyVacheck committed:
refactor + update: добавить в eslint правил

chyVacheck committed:
add component: SearchForm

1. Создать компонент Поисковой формы
2. Сделать адаптивную верстку к компоненту

chyVacheck committed:
add component: MoviesCard

chyVacheck committed:
add component: MoviesCardList

chyVacheck committed:
add component: MoreButton

chyVacheck committed:
add component: Movies

chyVacheck committed:
style code: подправить код в header под один стиль

chyVacheck committed:
add component: SavedMovies

chyVacheck committed:
feat + refactor: сделать закрытие меню после нажатии на кнопку

chyVacheck committed:
add component: Profile

chyVacheck committed:
style + feat: сделать адаптив компонента signform

1. Сделать адаптив компонента signform (стр регистрации, авторизации)
2. привести к стилю eslint

chyVacheck committed:
refactor: использовать контекст вместо пропсов

chyVacheck committed:
add component: Preloader

chyVacheck committed:
feat: сохранять значение фильтра короткометражек

Теперь в локальном хранилище сохраняется данные о фильтра короткометражек, при загрузке страницы данные берутся от туда

chyVacheck committed:
feat: сделать "якоря" и привязать их к нав. меню

chyVacheck committed:
feat: добавить проверку локального хранилища

Добавить проверку, что бы сайт не падал при неправильно заполненным локальном хранилищем

coolswood commented:
Работа отклоняется без потери итерации, так как не загружена ссылка на макет в README.md

chyVacheck committed:
Update README.md

coolswood commented:
Работа отклоняется без потери итерации, так как проект собирается с ошибками.
<img width="1350" alt="Screenshot 2023-04-13 at 08 41 02" src="https://user-images.githubusercontent.com/25247802/231664029-fda15fac-d763-4989-9b12-0b95ce81983c.png">

chyVacheck committed:
fix: убрать ошибки eslint

chyVacheck committed:
feat: сделать кнопку сохранения данных

1. Сделать кнопку сохранения данных
2. Сделать стили для отключенной кнопки

chyVacheck committed:
Update package-lock.json

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`class="portfolio-web-site" это должна быть единая ссылка (при клике и на текст и на иконку)`

- [x] <!--5.55--> Именование классов сделаны по БЭМ.
`Есть ошибки в нейминге, проверять здесь (https://nglazov.github.io/bem-validator-page/)`

- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
`Не установлена максимальная ширина контента`
`Необходимо проверить валидацию (https://validator.w3.org/)`

- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
`Перечисляемые элементы следует размечать списками class="portfolio-web-site"`
`class="SearchForm__field-input" это должна быть форма`

- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;x
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:ф
	- [x] <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы формы поиска фильмов не выделяются, когда на них установлен ФОКУС.`

- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`У формы поиска фильмов нет свойства required`

- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 72

## Доп. Комментарии:

chyVacheck committed:
fix+ feat: исправить семантику + добавить стили

1. Сделать "обуртку" инпута формой
2. Добавить стили для состояния "фокуса" на инпуте

chyVacheck committed:
delete: удалить не нужный класс

chyVacheck committed:
fix: исправить БЭМ ошибки

chyVacheck committed:
feat: установить максимальную ширину

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
`Необходимо проверить валидацию (https://validator.w3.org/)`

- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;x
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:ф
	- [x] <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 97

## Доп. Комментарии:

chyVacheck commented:
# Третий этап

Это третий этап моего диплома, в котором были сделаны большие усилия над `frontend` частью этого проекта, по сути половина, а именно верстка

## Сделано
- Полная верстка по макету
- Минимальная логика, для работы input`ов, а так же кнопок

## Остается
- 4 этап диплома - интеграцию с `backend`
- 5 этап, личная доработка
